# Alyssa

### Milestone 4

In [ ]:
#import 
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


### Research question
**What is the relationship between the active participation of music in one's lifestyle (composing, playing an instrument, listening to music on a greater than average amount, or neither) in relation to the severity of mental health disorders among these individuals? How do these factors influence how they percieve the effect of music?**

**Analysis plan:** <br>

To Do:
- [x] clean data -> check for any missing values and if there are any incorrect data that is beyond the range of interest
- [x] Columns: instrumentalist, composer, hours per day, anxiety, depression, OCD, and insomnia, music effect
- [x] Sort the instrumentalist and composer columns into one column: type of music participation
    - with the categories: instrumentalist, composer, both, neither
- [x] Creating a new column to determine whether an individual listens to music on a greater than average or lower than average amount, based on the mean provided in this data
- [x] add an overall mental health analysis in the form of a percentage
- [x] setting anxiety and depression to be 35% each and insomnia to be 20%, and OCD to be 10%



Creating the dataframe 

In [ ]:
dataset=pd.read_csv('../data/raw/music_therapy_and_mental_health_survey_results (1).csv')
dataset=dataset.dropna().reset_index(drop=True)

Subsetting the proper columns

In [ ]:
dataset=dataset[['Instrumentalist', 'Composer', 'Hours per day', 'Anxiety', 'Depression', 'Insomnia', 'OCD','Music effects']]


First, the hours per day must be edited into data based on a percentage of how much they listen to music compared to the average. To do this, the apply function is used to standardize the data. We previously found in Milestone 3 that the average number of hours people listen to music is 3.57 and that some people said they listened to music 24 hours a day.

In [ ]:
dataset['Hours per day'].max()


24.0

In [ ]:
# given that it is basically impossible to listen to music more than 20 hours a day, I will remove these datapoints
dataset[dataset['Hours per day']>=20].index.values

array([ 13, 307, 579], dtype=int64)

In [ ]:
dataset=dataset.drop([13,307,579]).reset_index(drop=True)
dataset

Instrumentalist Composer  Hours per day  Anxiety  Depression  Insomnia  \
0                No       No            4.0      7.0         7.0      10.0   
1                No      Yes            2.5      9.0         7.0       3.0   
2                No       No            4.0      7.0         2.0       5.0   
3               Yes      Yes            5.0      8.0         8.0       7.0   
4               Yes       No            3.0      4.0         8.0       6.0   
..              ...      ...            ...      ...         ...       ...   
608             Yes       No            2.0      7.0         6.0       0.0   
609             Yes       No            1.0      3.0         2.0       2.0   
610              No      Yes            6.0      2.0         2.0       2.0   
611             Yes       No            5.0      2.0         3.0       2.0   
612              No       No            2.0      2.0         2.0       2.0   

     OCD Music effects  
0    2.0     No effect  
1    3.0       Improve  
2    9.0       Improve  
3    7.0       Improve  
4    0.0       Improve  
..   ...           ...  
608  9.0       Improve  
609  5.0       Improve  
610  2.0       Improve  
611  1.0       Improve  
612  5.0       Improve  

[613 rows x 8 columns]

In [ ]:
dataset['Hours per day (%)'] = dataset['Hours per day'].apply(lambda x: ((x-dataset['Hours per day'].mean())/(dataset['Hours per day'].std())+1)/2*100)
dataset

Instrumentalist Composer  Hours per day  Anxiety  Depression  Insomnia  \
0                No       No            4.0      7.0         7.0      10.0   
1                No      Yes            2.5      9.0         7.0       3.0   
2                No       No            4.0      7.0         2.0       5.0   
3               Yes      Yes            5.0      8.0         8.0       7.0   
4               Yes       No            3.0      4.0         8.0       6.0   
..              ...      ...            ...      ...         ...       ...   
608             Yes       No            2.0      7.0         6.0       0.0   
609             Yes       No            1.0      3.0         2.0       2.0   
610              No      Yes            6.0      2.0         2.0       2.0   
611             Yes       No            5.0      2.0         3.0       2.0   
612              No       No            2.0      2.0         2.0       2.0   

     OCD Music effects  Hours per day (%)  
0    2.0     No effect          57.268822  
1    3.0       Improve          29.797873  
2    9.0       Improve          57.268822  
3    7.0       Improve          75.582788  
4    0.0       Improve          38.954856  
..   ...           ...                ...  
608  9.0       Improve          20.640890  
609  5.0       Improve           2.326925  
610  2.0       Improve          93.896753  
611  1.0       Improve          75.582788  
612  5.0       Improve          20.640890  

[613 rows x 9 columns]

In [ ]:
def type_of_music(x):
    '''
    Decides what type of music the responder participated in. If they were a composer, instrumentalist, both, or neither
    '''
    if (x['Instrumentalist']=="Yes") & (x['Composer']=="Yes"):
        return "Both"
    elif x['Instrumentalist']=="Yes": 
        return "Instrumentalist"  
    elif x['Composer']=="Yes":
        return "Composer"
    else:
        return "Neither"

In [ ]:
dataset['Music Participation']= dataset.apply(type_of_music, axis="columns")
dataset

Instrumentalist Composer  Hours per day  Anxiety  Depression  Insomnia  \
0                No       No            4.0      7.0         7.0      10.0   
1                No      Yes            2.5      9.0         7.0       3.0   
2                No       No            4.0      7.0         2.0       5.0   
3               Yes      Yes            5.0      8.0         8.0       7.0   
4               Yes       No            3.0      4.0         8.0       6.0   
..              ...      ...            ...      ...         ...       ...   
608             Yes       No            2.0      7.0         6.0       0.0   
609             Yes       No            1.0      3.0         2.0       2.0   
610              No      Yes            6.0      2.0         2.0       2.0   
611             Yes       No            5.0      2.0         3.0       2.0   
612              No       No            2.0      2.0         2.0       2.0   

     OCD Music effects  Hours per day (%) Music Participation  \
0    2.0     No effect          57.268822             Neither   
1    3.0       Improve          29.797873            Composer   
2    9.0       Improve          57.268822             Neither   
3    7.0       Improve          75.582788                Both   
4    0.0       Improve          38.954856     Instrumentalist   
..   ...           ...                ...                 ...   
608  9.0       Improve          20.640890     Instrumentalist   
609  5.0       Improve           2.326925     Instrumentalist   
610  2.0       Improve          93.896753            Composer   
611  1.0       Improve          75.582788     Instrumentalist   
612  5.0       Improve          20.640890             Neither   

     Overall Mental Health  
0                     7.10  
1                     6.50  
2                     5.05  
3                     7.70  
4                     5.40  
..                     ...  
608                   5.45  
609                   2.65  
610                   2.00  
611                   2.25  
612                   2.30  

[613 rows x 11 columns]

Creating an overall mental health column based on anxiety being 35%, depression being 35%, insomnia being 20% and OCD being 10%

In [ ]:
dataset['Overall Mental Health']=dataset['Anxiety']*.35+dataset['Depression']*.35+dataset['Insomnia']*.2+dataset['OCD']*.1
dataset

Instrumentalist Composer  Hours per day  Anxiety  Depression  Insomnia  \
0                No       No            4.0      7.0         7.0      10.0   
1                No      Yes            2.5      9.0         7.0       3.0   
2                No       No            4.0      7.0         2.0       5.0   
3               Yes      Yes            5.0      8.0         8.0       7.0   
4               Yes       No            3.0      4.0         8.0       6.0   
..              ...      ...            ...      ...         ...       ...   
608             Yes       No            2.0      7.0         6.0       0.0   
609             Yes       No            1.0      3.0         2.0       2.0   
610              No      Yes            6.0      2.0         2.0       2.0   
611             Yes       No            5.0      2.0         3.0       2.0   
612              No       No            2.0      2.0         2.0       2.0   

     OCD Music effects  Hours per day (%) Music Participation  \
0    2.0     No effect          57.268822             Neither   
1    3.0       Improve          29.797873            Composer   
2    9.0       Improve          57.268822             Neither   
3    7.0       Improve          75.582788                Both   
4    0.0       Improve          38.954856     Instrumentalist   
..   ...           ...                ...                 ...   
608  9.0       Improve          20.640890     Instrumentalist   
609  5.0       Improve           2.326925     Instrumentalist   
610  2.0       Improve          93.896753            Composer   
611  1.0       Improve          75.582788     Instrumentalist   
612  5.0       Improve          20.640890             Neither   

     Overall Mental Health  
0                     7.10  
1                     6.50  
2                     5.05  
3                     7.70  
4                     5.40  
..                     ...  
608                   5.45  
609                   2.65  
610                   2.00  
611                   2.25  
612                   2.30  

[613 rows x 11 columns]

### Method Chaining

Doing the same thing as the previous except using method chaining

In [ ]:
imean=dataset['Hours per day'].mean()
istd=dataset['Hours per day'].std()
print(imean)
print(istd)

3.6030995106035886
2.730156903404203


In [ ]:
def standardization(x):
    '''
    returning the standardized value of Hours per day based on the maximum value, standard deviation, and the mean of the dataset
    '''
    return ((x['Hours per day']-3.6030995106035886)/(2.730156903404203)+1)/8.470962736230923*100


In [ ]:
chaindf=(
    pd.read_csv('../data/raw/music_therapy_and_mental_health_survey_results (1).csv')
    .dropna()
    .assign(
	Overall_mental_health=lambda x: x['Anxiety']*.35+x['Depression']*.35+x['Insomnia']*.2+x['OCD']*.1)
    .assign(
        Music_Participation=lambda x: x.apply(type_of_music, axis="columns"))
    .reset_index(drop=True)
    .assign(
        Hours_per_day_Percentage=lambda x: x.apply(standardization,axis="columns")
    )
    .loc[:, ["Instrumentalist", "Composer", "Hours_per_day_Percentage", "Anxiety", "Depression","Insomnia","OCD", "Music effects","Music_Participation","Overall_mental_health"]]
)
chaindf

Instrumentalist Composer  Hours_per_day_Percentage  Anxiety  Depression  \
0                No       No                 13.521207      7.0         7.0   
1                No      Yes                  7.035298      9.0         7.0   
2                No       No                 13.521207      7.0         2.0   
3               Yes      Yes                 17.845147      8.0         8.0   
4               Yes       No                  9.197268      4.0         8.0   
..              ...      ...                       ...      ...         ...   
611             Yes       No                  4.873328      7.0         6.0   
612             Yes       No                  0.549388      3.0         2.0   
613              No      Yes                 22.169087      2.0         2.0   
614             Yes       No                 17.845147      2.0         3.0   
615              No       No                  4.873328      2.0         2.0   

     Insomnia  OCD Music effects Music_Participation  Overall_mental_health  
0        10.0  2.0     No effect             Neither                   7.10  
1         3.0  3.0       Improve            Composer                   6.50  
2         5.0  9.0       Improve             Neither                   5.05  
3         7.0  7.0       Improve                Both                   7.70  
4         6.0  0.0       Improve     Instrumentalist                   5.40  
..        ...  ...           ...                 ...                    ...  
611       0.0  9.0       Improve     Instrumentalist                   5.45  
612       2.0  5.0       Improve     Instrumentalist                   2.65  
613       2.0  2.0       Improve            Composer                   2.00  
614       2.0  1.0       Improve     Instrumentalist                   2.25  
615       2.0  5.0       Improve             Neither                   2.30  

[616 rows x 10 columns]

In [ ]:
chaindf["Hours_per_day_Percentage"].max()

100.0

Also, for the sake of summarizing the varying number of activities that people participate in music and comparing it to the average of the overall mental health of the categories


In [ ]:
chaindf.groupby("Music_Participation").mean()

C:\Users\alkon\AppData\Local\Temp\ipykernel_13652\1788567879.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  chaindf.groupby("Music_Participation").mean()


Hours_per_day_Percentage   Anxiety  Depression  Insomnia  \
Music_Participation                                                             
Both                                16.334373  6.144578    5.518072  4.578313   
Composer                            17.412753  5.320000    5.080000  4.160000   
Instrumentalist                      8.652156  6.000000    4.448718  3.679487   
Neither                             12.105145  5.829923    4.883632  3.649616   

                          OCD  Overall_mental_health  
Music_Participation                                   
Both                 3.000000               5.297590  
Composer             2.200000               4.692000  
Instrumentalist      2.653846               4.658333  
Neither              2.617647               4.741432

In [ ]:
#wrapping it in a function
def load_and_process(url_or_path_to_csv_file):

    # Method Chain 1 (Load data and deal with missing data)

    df1 = (
        pd.read_csv(url_or_path_to_csv_file)
        .dropna()
        .reset_index(drop=True)
)

    # Method Chain 2 (Create new columns, drop others, and do processing)

    df2 = (
          df1
        .assign(
            Overall_mental_health=lambda x: x['Anxiety']*.35+x['Depression']*.35+x['Insomnia']*.2+x['OCD']*.1)
        .assign(
            Music_Participation=lambda x: x.apply(type_of_music, axis="columns"))
        .reset_index(drop=True)
        .assign(
            Hours_per_day_Percentage=lambda x: x.apply(standardization,axis="columns")
        )
        .loc[:, ["Instrumentalist", "Composer", "Hours_per_day_Percentage", "Anxiety", "Depression","Insomnia","OCD", "Music effects","Music_Participation","Overall_mental_health"]]


      )

    # Make sure to return the latest dataframe

    return df2 

df=load_and_process('../data/raw/music_therapy_and_mental_health_survey_results (1).csv')
df

Instrumentalist Composer  Hours_per_day_Percentage  Anxiety  Depression  \
0                No       No                 13.521207      7.0         7.0   
1                No      Yes                  7.035298      9.0         7.0   
2                No       No                 13.521207      7.0         2.0   
3               Yes      Yes                 17.845147      8.0         8.0   
4               Yes       No                  9.197268      4.0         8.0   
..              ...      ...                       ...      ...         ...   
611             Yes       No                  4.873328      7.0         6.0   
612             Yes       No                  0.549388      3.0         2.0   
613              No      Yes                 22.169087      2.0         2.0   
614             Yes       No                 17.845147      2.0         3.0   
615              No       No                  4.873328      2.0         2.0   

     Insomnia  OCD Music effects Music_Participation  Overall_mental_health  
0        10.0  2.0     No effect             Neither                   7.10  
1         3.0  3.0       Improve            Composer                   6.50  
2         5.0  9.0       Improve             Neither                   5.05  
3         7.0  7.0       Improve                Both                   7.70  
4         6.0  0.0       Improve     Instrumentalist                   5.40  
..        ...  ...           ...                 ...                    ...  
611       0.0  9.0       Improve     Instrumentalist                   5.45  
612       2.0  5.0       Improve     Instrumentalist                   2.65  
613       2.0  2.0       Improve            Composer                   2.00  
614       2.0  1.0       Improve     Instrumentalist                   2.25  
615       2.0  5.0       Improve             Neither                   2.30  

[616 rows x 10 columns]

In [ ]:
import scripts

In [ ]:
from scripts import project_functions1 as pt

In [ ]:
import scripts
from scripts import project_functions1 as pt
df=pt.load_and_process('../data/raw/music_therapy_and_mental_health_survey_results (1).csv')
df

NameError: name 'type_of_music' is not defined

Visualizing the question through various dataset. The higher the overall mental health indicates more severe outcomes.

In [ ]:
sns.set_theme(style="white", rc={"axes.facecolor": (0, 0, 0, 0)}, font_scale=2
)

# Initialize the FacetGrid object
pal = sns.cubehelix_palette(10, rot=-.25, light=.7)
g = sns.FacetGrid(df, row="Music_Participation", hue="Music_Participation", aspect=8, height=2, palette="mako")

# Draw the densities 
g.map(sns.kdeplot, "Overall_mental_health",
      bw_adjust=.5, clip_on=False,
      fill=True, alpha=1, linewidth=1.5)
g.map(sns.kdeplot, "Overall_mental_health", color="w", lw=2, bw_adjust=.5)

# passing color=None to refline() uses the hue mapping
g.refline(y=0, linewidth=2, linestyle="-", color=None, clip_on=False)


# Define and use a simple function to label the plot in axes coordinates
def label(x, color, label):
    ax = plt.gca()
    ax.text(-0.1, 0.2, label, fontweight="bold", color=color,
            ha="left", va="center", transform=ax.transAxes)


g.map(label, "Overall_mental_health")

# Set the subplots to not overlap
g.figure.subplots_adjust(hspace=.8)

# Remove axes details that don't play well with overlap
g.set_titles("")
g.set(yticks=[], xlabel="Overall Mental Health", ylabel="")

g.despine(bottom=True, left=True)
g.set(xlim=(0, 10))
plt.suptitle("Effects of music participation on mental health", y=0.98)


Nothing too conclusive can be drawn from this - the distribution is pretty centralized towards 4 and 6. I will try a different plot to see if I can visualize the differences better.

In [ ]:
#Set theme
sns.set_theme(style="white", font_scale=1
)

# Initialize the figure
f, ax = plt.subplots()
sns.despine(bottom=True, left=True)

# Show each observation with a scatterplot
sns.stripplot(
    data=df, x="Overall_mental_health", y="Music_Participation", hue="Music_Participation",
    dodge=True, alpha=.25, zorder=1, legend=False
)

# Show the conditional means, aligning each pointplot in the
# center of the strips by adjusting the width allotted to each
# category (.8 by default) by the number of hue levels
sns.pointplot(
    data=df, x="Overall_mental_health", y="Music_Participation", hue="Music_Participation",
    join=False, dodge=.8 - .8/ 3, palette="dark",
    markers="d", scale=.8, errorbar=None
)

# Improve the legend
sns.move_legend(
    ax, loc="upper right", bbox_to_anchor=(1.4, 1) 
)

#titles
plt.suptitle("Effects of music participation on mental health", y=0.98)
plt.xlabel("Overall mental health 0-10") 
plt.ylabel("Music Participation") 


This visualization is also very difficult to tell what the differences are. I will experiment with a different distribution.

In [ ]:

# Draw a violinplot and striplot
sns.violinplot(data=df, x="Overall_mental_health", y="Music_Participation", palette="mako")
sns.stripplot(data=df, x='Overall_mental_health', y='Music_Participation', color='white', alpha=0.5)

# Finalize the figure
ax.set(ylim=(-1, 1.05))
ax.set(xlim=(-1, 11))
sns.despine(left=True, bottom=True)

#titles
plt.suptitle("Effects of music participation on mental health", y=0.98)
plt.xlabel("Overall mental health 0-10") 
plt.ylabel("Music Participation") 

# the TA likes this graph the most


To be honest, all of these visualizations demonstrate that there really isn't any correlation with music participation and overall mental health. I can check for hours per day against overall mental health if there is any other form of correlation though - I will check with both the percentage calculation and the non-processed hours per day data.

In [ ]:
ax=sns.jointplot(data=df, x="Hours_per_day_Percentage", y="Overall_mental_health", kind="hex", color="#4CB391") 
ax.set_axis_labels("Hours per day %", "Overall Mental Health", fontsize=16)
plt.suptitle("Hours per day of music played in correspondence to mental health", y=1.04)

#choose only one 

In [ ]:

ax1=sns.jointplot(data=dataset, x="Hours per day", y="Overall Mental Health", kind="hex", color="#4CB391")
ax1.set_axis_labels("Hours per day (h)", "Overall Mental Health", fontsize=16)
plt.suptitle("Hours per day of music played in correspondence to mental health", y=1.04)



There doesn't seem to be much of a positive correlation with the hours per day either. I will check whether music participation can influence the percieved effect of music however. This will require a bit of data wrangling. 

In [ ]:
dfnew=df[['Music_Participation', 'Music effects']]
dfnew=pd.DataFrame(dfnew.groupby(['Music effects','Music_Participation'])['Music_Participation'].count())
dfnew

In [ ]:
df2=df.groupby(['Music effects','Music_Participation']).size().unstack('Music_Participation', fill_value=0)
df2

Since there are different amounts, I will probably have to convert the frequencies into percentages out of a whole

In [ ]:
list(df2.columns)


In [ ]:
df3=(df2
    .assign(
        Both_P=lambda x: x['Both']/x['Both'].sum()*100)
    .assign(
        Composer_P=lambda x: x['Composer']/x['Composer'].sum()*100)
    .assign(
        Instrumentalist_P=lambda x: x['Instrumentalist']/x['Instrumentalist'].sum()*100)
     .assign(
        Neither_P=lambda x: x['Neither']/x['Neither'].sum()*100)
    .loc[:, ['Both_P','Composer_P','Instrumentalist_P','Neither_P']]
)
df3
df4= df3.transpose()
df4

In [ ]:
sns.set_theme(style="white", font_scale=1
)

#set y labels
# get y labels to be straight instead of sideways 
y_axis_labels = ["Both", "Composer", "Instrumentalist", "Neither"] 

sns.heatmap(data=df4, cmap="crest",  yticklabels=y_axis_labels)
plt.suptitle("Type of Music Pariticpation and percieved effects of music", y=1.003)
plt.ylabel("Music Participation")
plt.xlabel("Music Effects in percentage of each type of category (%)")

Based on these results, the percieved improvement is higher for composers compared to the other categories. Almost all the categories saw that the music had a positive music effect.

### Research Questions for Milestone 3

**What is the relationship between the active participation of music in one's lifestyle (composing, playing an instrument, listening to music on a greater than average amount, or neither) in relation to the severity of mental health disorders among these individuals?**

**Analysis plan:**
For my analysis, I plan on using the columns instrumentalist, composer, hours per day, anxiety, depression, OCD, and insomnia. I plan on sorting the instrumentalist and composer columns into one column: type of music participation, with the categories: instrumentalist, composer, both, neither. I also plan on creating a new column to determine whether an individual listens to music on a greater than average or lower than average amount, based on the mean provided in this data. I will then analyze the means of different anxiety disorders against each of these new columns with categories and visualize the data using a barplot or boxplot. I will clean my data by checking for any missing values and if there are any incorrect data that is beyond the range of interest. If there are any outliers, I will indicate as such and deal with these appropriately. I will also add an overall mental health analysis in the form of a percentage by doing some further data manipulation, by setting anxiety and depression to be 35% each and insomnia to be 20%, and OCD to be 10%. Finally, I will visualize said data in the most appropriate manner after a few different experimentations. I may view the frequency of hours of music listened per day and a scatterplot heatmap of each of the different disorder categories.

<br>

**What is the correlation between an instrument player, a composer, neither, or frequency of music listened per day and how they percieve the effect of music on their lives?**

 <br>**Analysis plan:**
Similar to before, I will most likely use the same dataframe with the newly created columns to analyze this data, with an extra column to include the music effects. I will have to remove any of the NaN values of music effects and check any of the other columns for incorrect or missing data. I will also plan on dealing with any outliers and noting this as well. After this processing, I will find a suitable method for visualizing the data. I will most likely use a similar method of barplots to determine the percentage distribution of percieved mental health effects against the different categories.




### Milestone 2: Research question/interests

**What is the relationship between the active participation of music in one's lifestyle (composing, playing an instrument, or neither) in relation to the frequency of anxiety and depression among these individuals?**
 
I plan on answering my first research question using the columns: instrumentalist, composer, anxiety and depression. I may visualize this data using a scatterplot based on categories of composing, playing an instrument or neither in relation to anxiety or depression and the scale of anxiety or depression on the y axis.  <br><br>

**Analysis plan:**
For my analysis, I will incorporate the columns: instrumentalist, composer, neither, anxiety, and depression into a data frame. I will clean my data by checking for any missing values and if there are any incorrect data that is beyond the range of interest. If there are any outliers, I will indicate as such and deal with these appropriately. I plan on adding an additional column "neither" to count the number of those who are not composing or playing an instrument by doing groupBy. I may also add an overall mental health analysis in the form of a percentage by doing some further data manipulation. Finally, I will visualize said data in the most appropriate manner after a few different experimentations.

<br>

**What is the correlation between an instrument player, a composer, or neither and how they percieve the effect of music on their lives?**

For my second research question, I plan on using the columns: instrumentalist, composer, and music effects. <br><br>
**Analysis plan:**
I will incorporate the columns: instrumentalist, composer, and music effects into a data frame. Once again, I will create a new aggregate column of those who are neither an instrumentalist or a composer. I will have to remove any of the NaN values of music effects and check any of the other columns for incorrect or missing data. I will also plan on dealing with any outliers and noting this as well. I will count the total number of those who have felt that music has had an improvement and compare it between those who are a composer, instrumentalist or neither. After this processing, I will find a suitable method for visualizing the data.




## Milestone 3: EDA & Research Questions 

In [ ]:
#import 
import pandas as pd
import seaborn as sns
#creating dataFrame
dataset = pd.read_csv('../data/raw/music_therapy_and_mental_health_survey_results (1).csv')

dataset.head()

In [ ]:
# Determining all the names of the columns in the dataset

pd.DataFrame(dataset.columns)

In [ ]:
# Figuring out the total rows and columns of the dataset (rows, columns)

dataset.shape

In [ ]:
# Since I am more interested in the columns: instrumentalist, composer, anxiety, depression, insomnia, OCD, and music effects, I'm going to produce a new dataframe with these columns

In [ ]:
df=dataset[['Instrumentalist', 'Composer', 'Anxiety', 'Depression', 'Insomnia', 'OCD', 'Music effects']]
df

In [ ]:
# Looking at the possible values of music effects

df['Music effects'].unique()

In [ ]:
#looking at how many NaN values are in column 'Music effects'

df['Music effects'].isna().sum()

In [ ]:
# Dropping rows that have NaN values to ensure that this does not interfere with the results
df=df.dropna()

In [ ]:
# Examining how many instrumentalists are in this dataset

(df['Instrumentalist']=='Yes').sum()

In [ ]:
# Examining how many composers are in this dataset
(df['Composer']=='Yes').sum()

In [ ]:
# Looking at the possible values of Instrumentalist

df['Instrumentalist'].unique()


In [ ]:
# Looking at the possible values of composer

df['Composer'].unique()

In [ ]:

# Looking at the highest and lowest values of Anxiety, depression, insomnia, OCD

pd.DataFrame([{"Anxiety":df['Anxiety'].min(),  "Depression":df['Depression'].min(),  "Insomnia":df['Insomnia'].min(), "OCD":df['OCD'].min()},{ "Anxiety":df['Anxiety'].max(), "Depression":df['Depression'].max(),  "Insomnia":df['Insomnia'].max(),  "OCD":df['OCD'].max()}])



In [ ]:
#examining the means, counts and other values of the numeric categories
df.describe()

*Notes:* <br>
Cell 3: There are 32 different variables to work with in this dataset. 

Cell 5: There seems to be three categories for music effects - no effect, improve, worsen.

Cell 7 & 8: There are 232 instrumentalists and 124 composers from a dataset of 735 responses.

Describe table: The average anxiety experience of 5.848 is higher than any other category of metnal health, this also has the least amount of standard deviation. The lowest experience of mental health seems to be OCD with a mean of 2.650. Depression is the second most frequent and insomnia is the third, and they both have a similar standard deviation. 

In [ ]:
# visualizing scatterplot of composers and instrumentalists related to their anxiety and depression, 0 being none and 10 being extreme
g= sns.FacetGrid(df, col="Composer", row="Instrumentalist")
g.map(sns.histplot, "Depression", "Anxiety")


*Notes:* <br>
From this data visualization, there seems to be the greatest amount of anxiety and depression among those who are neither composers or instrumentalists and there are the least amount of anxiety and depression among those who are both instrumentalists and composers. Despite this, the data is fairly distributed so it is extremely difficult to determnine whether this relationship is significant and more data visualizations will have to be done to get a better visualization. There also seems to be the issue of lack of data for the composer category.

In [ ]:
# visualizing the distribution of different mental health disorders
ax1=sns.boxplot(data=df).set(xlabel="Mental Health disorder",ylabel="Degree of severity, 10 being the most")



*Notes:* <br>
From this data visualization, you can see that anxeity was reported to be the most severe, whereas OCD was the lowest. In the order of anxiety, depression, insomnia, and OCD were the most to least severe. Anxiety also had the smallest standard deviation whereas the other categories were fairly similar.

In [ ]:
# Seeing the music effects for instrumentalists -- determining the distribution of improvement, no effect, or worsening effect among instrumentalists vs noninstrumentalists
df1 = df.groupby('Instrumentalist')['Music effects'].value_counts(normalize=True)
df1 = df1.mul(100)
df1 = df1.rename('percent').reset_index()
g = sns.catplot(x='Music effects',y='percent',hue='Instrumentalist',kind='bar',data=df1)
g.ax.set_ylim(0,100)

In [ ]:
# Seeing the music effects for composers -- determining the distribution of improvement, no effect, or worsening effect among composers vs noncomposers
df1 = df.groupby('Composer')['Music effects'].value_counts(normalize=True)
df1 = df1.mul(100)
df1 = df1.rename('percent').reset_index()
g = sns.catplot(x='Music effects',y='percent',hue='Composer',kind='bar',data=df1)
g.ax.set_ylim(0,100)


*Notes:* <br>
From either visualization, the distribution of music effects for composers and instrumentalists skewed more heavily towards positive effects compared to those who did not participate as actively in music. Despite this, a great amount of people who did not participate in either activity still felt as though music had a positive effect on their mental wellbeing. I wonder if this is because even if they did not actively participate in music, they still actively listened to it on a daily basis. As a result, I might pivot my research question to factor in the variable of time spent per day listening to music.

In [ ]:
#determining the distribution of music listened per day.

dataset['Hours per day'].describe()

In [ ]:
# visualizing the distribution of music listened per day

sns.countplot(y=dataset["Hours per day"])

# sswitch x and y axis - feedback